### PR TITLE
feat(pv): `iSCSI` Persistent Volume and possibility to bind PVs to PVCs in any namespace

### PR DESCRIPTION
**Pull request by St0rmingBr4in**
This adds the IscsiPersistentVolume class to be able to generate iSCSI backed persistent volumes.

This also adds the ability to bind a PV to a PVC in another namespace than the default one.

### CODE REVIEW

**Comment by St0rmingBr4in** [comment]
@iliapolo Could you take a look please ?

File: src/pv.ts [hunk 1]
```diff
@@ -208,10 +209,10 @@ export class PersistentVolume extends base.Resource implements IPersistentVolume
    *
    * @see https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reserving-a-persistentvolume
    */
-  public reserve(): pvc.PersistentVolumeClaim {
+  public reserve(namespace?: string): pvc.PersistentVolumeClaim {
```
Comment (iliapolo) [inline src/pv.ts:212]: We don't normally use optional positional arguments. Instead, create a `PersistentVolumeReserveOptions` interface.

Also this requires a test.

File: src/pv.ts [hunk 2]
```diff
@@ -208,10 +209,10 @@ export class PersistentVolume extends base.Resource implements IPersistentVolume
...
     const claim = new pvc.PersistentVolumeClaim(this, `${this.name}PVC`, {
-      metadata: { name: `pvc-${this.name}`, namespace: this.metadata.namespace },
-
+      metadata: { name: `pvc-${this.name}` },
+      namespace: namespace ? namespace : this.metadata.namespace,
```
Comment (iliapolo) [inline src/pv.ts:215]:
```suggestion
      namespace: namespace ?? this.metadata.namespace,
```

**Review by iliapolo — CHANGES_REQUESTED** [review]

**Comment by iliapolo** [comment]
Awesome, thanks for this!

**Pull request closed without merging**